pkg/csi/service/vanilla/controller.go

```diff
@@ -2674,6 +2697,19 @@ func (c *controller) processQueryResultsListVolumes(ctx context.Context, startin
 	return entries, nextToken, volumeType, nil
 }
 
+// isVolumeMigrated checks if the given volume is a migrated in-tree volume
+func isVolumeMigrated(ctx context.Context, volumeID string) (bool, error) {
+	// todo: confirm this assumption with team.
+	//  volumeIDToNameMap stores key as volumeID for CSI volumes and volumePath for in-tree migrated volumes
+	//  since we don't have the path (need PV obj for that), we can assume that if a volumeID is not present in this map,
+	//  it must be an in-tree volume
+	pvcNamespace, ok := commonco.ContainerOrchestratorUtility.GetPVCNamespaceFromVolumeID(ctx, volumeID)
```

Comment:
How retrieving namespace confirm volume is in-tree or CSI?

Reply:
Discussed offline. Resolving.